### PR TITLE
Modernize the dav admin settings

### PR DESCRIPTION
**Pull request by CarlSchwan**
- Use nextcloud-vue components (with proper doc link now)
- Add underline to links so we can see them (accessibility++)

image

### CODE REVIEW

File: apps/dav/src/views/CalDavSettings.vue [hunk 1]
```diff
@@ -58,47 +50,62 @@
...
 				{{ $t('dav', 'Enable notifications for events via push') }}
-			</label>
+			</CheckboxRadioSwitch>
 		</p>
-	</div>
+	</SettingsSection>
 </template>
 
-<style lang="scss" scoped>
+<style scoped>
```
Comment (nickvergessen) [inline apps/dav/src/views/CalDavSettings.vue:74]: `The <script> should be above the <style> on line 74`

Mind to flip it?

File: apps/dav/src/views/CalDavSettings.vue [hunk 2]
```diff
@@ -1,35 +1,30 @@
 <template>
-	<div class="section">
-		<h2>{{ $t('dav', 'Calendar server') }}</h2>
+	<SettingsSection :title="$t('dav', 'Calendar server')"
+		:docUrl="userSyncCalendarsDocUrl">
```
Comment (nickvergessen) [inline apps/dav/src/views/CalDavSettings.vue:3]: Attribute ':docUrl' must be hyphenated     

```suggestion
		:doc-url="userSyncCalendarsDocUrl">
```

File: apps/dav/src/views/CalDavSettings.vue (hunk 1 above)
Comment (JuliaKirschenheuter) [reply apps/dav/src/views/CalDavSettings.vue:74]: > `The <script> should be above the <style> on line 74`

why do you think so? It is inside .vue - then it is all right

> Mind to flip it?

I would also say yes - linter would complain about it

File: apps/dav/src/views/CalDavSettings.vue (hunk 2 above)
Comment (JuliaKirschenheuter) [reply apps/dav/src/views/CalDavSettings.vue:3]: @CarlSchwan pleas make ``` npm run lint:fix``` and you would not have this problems

Comment (CarlSchwan) [reply apps/dav/src/views/CalDavSettings.vue:3]: eslint is always a pain when I have some apps in my repo

```
Oops! Something went wrong! :(

ESLint: 8.7.0

ESLint couldn't find the plugin "eslint-plugin-cypress".

(The package "eslint-plugin-cypress" was not found when loaded as a Node module from the directory "/home/cschwan/nextcloud/workspace/server/apps-extra/activity/cypress".)

It's likely that the plugin isn't installed correctly. Try reinstalling by running the following:

    npm install eslint-plugin-cypress@latest --save-dev

The plugin "eslint-plugin-cypress" was referenced from the config file in "apps-extra/activity/cypress/.eslintrc.json".

If you still can't figure out the problem, please stop by https://eslint.org/chat/help to chat with the team.
```

Comment (CarlSchwan) [reply apps/dav/src/views/CalDavSettings.vue]: Found a solution https://github.com/nextcloud/server/pull/32558

**Review by ChristophWurst — APPROVED** [review]

**Comment by nickvergessen** [comment]
Node and Note / Test seem broken?

**Comment by CarlSchwan** [comment]
acceptance-app-files-sharing are half of the time broken even without this PR :(

**Review by JuliaKirschenheuter — APPROVED** [review]
@CarlSchwan , good work, thank you!

I just have small comments to implementation of CheckboxRadioSwitch, but this is an Issue for NC-Vue, and actually should not be fixed there. I will open a new accessibility issue for NC-Vue.

**Pull request merged**